Yes, @quchenhao that's correct.

@GroovySmoothie, here is an example of a failure we observe in the PR testing:
```
AssertionError: Teardown of test_rollback_to_stable13.test_rollback_to_stable13.test_rollback_to_stable_with_aborted_updates -s 12 (row_integer.no_prepare.no_dryrun.0) failed with message: Traceback (most recent call last):
  File "wiredtiger/tools/rts_verifier/rts_verify.py", line 17, in <module>
    op = Operation(line)
  File "wiredtiger/tools/rts_verifier/operation.py", line 78, in __init__
    ptr(line)
  File "wiredtiger/tools/rts_verifier/operation.py", line 523, in __init_hs_update_remove
    durable_start_start = int(matches.group(1))
AttributeError: 'NoneType' object has no attribute 'group'
```

In [operation.py](https://github.com/wiredtiger/wiredtiger/blob/ada529c6e4db9e732d83296fc5a388933a6519d4/tools/rts_verifier/operation.py), we are expecting a specific format (see an example [here](https://github.com/wiredtiger/wiredtiger/blob/ada529c6e4db9e732d83296fc5a388933a6519d4/tools/rts_verifier/operation.py#L521)) that we need to update after your changes. To verify your changes, you can do the following:

```
cd build
python3 ../test/suite/run.py rollback_to_stable13
```

If you want to test all the rollback_to_stable tests, you can do:
```
# The -j value will depend on your platform
python3 ../test/suite/run.py -j $(sysctl -n hw.physicalcpu) -v 2 rollback_to_stable
```

Finally, we have some formatting/linting tools that you will need to run:

```
cd dist
./s_all # All tools are executed
./s_fast # Usually this is enough.
```

Please let us know if you have any questions.

--- a/src/support/timestamp.c
+++ b/src/support/timestamp.c
@@ -46,17 +46,18 @@ __wt_time_window_to_string(WT_TIME_WINDOW *tw, char *tw_string)
 {
     char ts_string[6][WT_TS_INT_STRING_SIZE];
 
-    WT_IGNORE_RET(__wt_snprintf(tw_string, WT_TIME_STRING_SIZE,
-      "start: durable_timestamp=%s timestamp=%s prepare_timestamp=%s prepared_id=%" PRIu64
-      " transaction=%" PRIu64 " | stop: durable_timestamp=%s timestamp=%s prepare_timestamp=%s "
-      "prepared_id=%" PRIu64 " transaction=%" PRIu64 "%s",
-      __wt_timestamp_to_string(tw->durable_start_ts, ts_string[0]),
-      __wt_timestamp_to_string(tw->start_ts, ts_string[1]),
-      __wt_timestamp_to_string(tw->start_prepare_ts, ts_string[2]), tw->start_prepared_id,
-      tw->start_txn, __wt_timestamp_to_string(tw->durable_stop_ts, ts_string[3]),
-      __wt_timestamp_to_string(tw->stop_ts, ts_string[4]),
-      __wt_timestamp_to_string(tw->stop_prepare_ts, ts_string[5]), tw->stop_prepared_id,
-      tw->stop_txn, WT_TIME_WINDOW_HAS_PREPARE(tw) ? ", prepared" : ""));
+    WT_IGNORE_RET(
+      __wt_snprintf(tw_string, WT_TIME_STRING_SIZE,
+        "start: durable_timestamp=%s timestamp=%s prepare_timestamp=%s prepared_id=%" PRIu64
+        " transaction=%" PRIu64 " | stop: durable_timestamp=%s timestamp=%s prepare_timestamp=%s "
+        "prepared_id=%" PRIu64 " transaction=%" PRIu64 "%s",
+        __wt_timestamp_to_string(tw->durable_start_ts, ts_string[0]),
+        __wt_timestamp_to_string(tw->start_ts, ts_string[1]),
+        __wt_timestamp_to_string(tw->start_prepare_ts, ts_string[2]), tw->start_prepared_id,
+        tw->start_txn, __wt_timestamp_to_string(tw->durable_stop_ts, ts_string[3]),
+        __wt_timestamp_to_string(tw->stop_ts, ts_string[4]),
+        __wt_timestamp_to_string(tw->stop_prepare_ts, ts_string[5]), tw->stop_prepared_id,
+        tw->stop_txn, WT_TIME_WINDOW_HAS_PREPARE(tw) ? ", prepared" : ""));
     return (tw_string);
 }
 

--- a/tools/rts_verifier/operation.py
+++ b/tools/rts_verifier/operation.py
@@ -54,6 +54,21 @@ class OpType(Enum):
     SKIP_DEL = 45,
     NO_STABLE = 46
 
+_TIME_WINDOW_REGEX = (
+    r'start: '
+    r'durable_timestamp=\((\d+), (\d+)\) '
+    r'timestamp=\((\d+), (\d+)\) '
+    r'prepare_timestamp=\((\d+), (\d+)\) '
+    r'prepared_id=(\d+) '
+    r'transaction=(\d+) '
+    r'\| stop: '
+    r'durable_timestamp=\((\d+), (\d+)\) '
+    r'timestamp=\((\d+), (\d+)\) '
+    r'prepare_timestamp=\((\d+), (\d+)\) '
+    r'prepared_id=(\d+) '
+    r'transaction=(\d+)'
+)
+
 class Operation:
     def __init__(self, line):
         self.line = line
@@ -297,7 +312,7 @@ class Operation:
         self.type = OpType.HS_UPDATE_ABORT
         self.file = self.__extract_file(line)
 
-        matches = re.search('start: \((\d+), (\d+)\)/\((\d+), (\d+)\)/\((\d+), (\d+)\)/(\d+)/(\d+) \| stop: \((\d+), (\d+)\)/\((\d+), (\d+)\)/\((\d+), (\d+)\)/(\d+)/(\d+)', line)
+        matches = re.search(_TIME_WINDOW_REGEX, line)
 
         durable_start_start = int(matches.group(1))
         durable_start_end = int(matches.group(2))
@@ -332,7 +347,7 @@ class Operation:
         self.type = OpType.HS_UPDATE_VALID
         self.file = self.__extract_file(line)
 
-        matches = re.search('start: \((\d+), (\d+)\)/\((\d+), (\d+)\)/\((\d+), (\d+)\)/(\d+)/(\d+) \| stop: \((\d+), (\d+)\)/\((\d+), (\d+)\)/\((\d+), (\d+)\)/(\d+)/(\d+)', line)
+        matches = re.search(_TIME_WINDOW_REGEX, line)
 
         durable_start_start = int(matches.group(1))
         durable_start_end = int(matches.group(2))
@@ -391,7 +406,7 @@ class Operation:
         self.type = OpType.HS_GT_ONDISK
         self.file = self.__extract_file(line)
 
-        matches = re.search('start: \((\d+), (\d+)\)/\((\d+), (\d+)\)/\((\d+), (\d+)\)/(\d+)/(\d+) \| stop: \((\d+), (\d+)\)/\((\d+), (\d+)\)/\((\d+), (\d+)\)/(\d+)/(\d+)', line)
+        matches = re.search(_TIME_WINDOW_REGEX, line)
 
         durable_start_start = int(matches.group(1))
         durable_start_end = int(matches.group(2))
@@ -430,7 +445,7 @@ class Operation:
         self.type = OpType.HS_STOP_OBSOLETE
         self.file = self.__extract_file(line)
 
-        matches = re.search('start: \((\d+), (\d+)\)/\((\d+), (\d+)\)/\((\d+), (\d+)\)/(\d+)/(\d+) \| stop: \((\d+), (\d+)\)/\((\d+), (\d+)\)/\((\d+), (\d+)\)/(\d+)/(\d+)', line)
+        matches = re.search(_TIME_WINDOW_REGEX, line)
         durable_start_start = int(matches.group(1))
         durable_start_end = int(matches.group(2))
         self.durable_start = Timestamp(durable_start_start, durable_start_end)
@@ -518,7 +533,7 @@ class Operation:
         self.stop = self.__extract_simple_timestamp('stop_timestamp', line)
         self.stable = self.__extract_simple_timestamp('stable_timestamp', line)
 
-        matches = re.search('start: \((\d+), (\d+)\)/\((\d+), (\d+)\)/\((\d+), (\d+)\)/(\d+)/(\d+) \| stop: \((\d+), (\d+)\)/\((\d+), (\d+)\)/\((\d+), (\d+)\)/(\d+)/(\d+)', line)
+        matches = re.search(_TIME_WINDOW_REGEX, line)
 
         durable_start_start = int(matches.group(1))
         durable_start_end = int(matches.group(2))

